iothub/device/src/DirectMethod/DirectMethodResponse.cs

```diff
@@ -39,7 +41,89 @@ public DirectMethodResponse(int status)
         /// </summary>
         protected internal PayloadConvention PayloadConvention { get; set; }
 
-        internal byte[] GetPayloadObjectBytes()
+        /// <summary>
+        /// The direct method response payload, deserialized to the specified type.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Use this method when the payload type is known and it can be deserialized using the configured
+        /// <see cref="PayloadConvention"/>. If it is not JSON or the type is not known, use <see cref="GetPayloadAsBytes"/>.
+        /// </para>
+        /// <para>
+        /// One intended usage of this method is to deserialize the direct method response received by an edge module client
+        /// after it invokes a direct method on an edge device or an edge module connected to the same edge hub.
+        /// These operations are invoked using the API <see cref="IotHubModuleClient.InvokeMethodAsync(string, EdgeModuleDirectMethodRequest, System.Threading.CancellationToken)"/>
+        /// and <see cref="IotHubModuleClient.InvokeMethodAsync(string, string, EdgeModuleDirectMethodRequest, System.Threading.CancellationToken)"/>.
+        /// </para>
+        /// </remarks>
+        /// <typeparam name="T">The type to deserialize the direct method response payload to.</typeparam>
+        /// <param name="payload">When this method returns true, this contains the value of the direct method response payload.
+        /// When this method returns false, this contains the default value of the type <c>T</c> passed in.</param>
+        /// <returns><c>true</c> if the direct method response payload can be deserialized to type <c>T</c>; otherwise, <c>false</c>.</returns>
+        /// <example>
+        /// <code language="csharp">
+        /// DirectMethodResponse response = await client
+        ///     .InvokeMethodAsync(deviceId, moduleId, directMethodRequest, cancellationToken)
+        ///     .ConfigureAwait(false);
+        /// if (response.TryGetPayload(out MyCustomType customTypePayload))
+        /// {
+        ///     // do work
+        ///     // ...
+        /// }
+        /// else
+        /// {
+        ///     // Get the payload as bytes
+        ///     byte[] arr = directMethodRequest.GetPayloadAsBytes();
+        ///
+        ///     // deserialize as needed and do work...
+        /// }
+        ///
+        /// // ...
+        /// </code>
+        /// </example>
+        public bool TryGetPayload<T>(out T payload)
```

Comment:
Previously only the service client had `TryGet...()` for the direct method response. We had not added them on the client side since a client would create and send the direct method response, so it didn't need helpers for getting the payload out.
However, since the module client is also expected to be working with direct method response (on the receiving end), I'm adding these methods in.

Reply:
Update the migration guide?